listenbrainz/listenstore/tests/test_influxlistenstore.py

```diff
@@ -238,6 +241,128 @@ def test_incremental_dump(self):
+
+        sleep(1)
+        self.assertTrue(os.path.isfile(dump_location))
+        self.reset_influx_db()
+        sleep(1)
+        self.logstore.import_listens_dump(dump_location)
+        sleep(1)
+        listens = self.logstore.fetch_listens(user_name=self.testuser_name, to_ts=11)
+        self.assertEqual(len(listens), 5)
+        self.assertEqual(listens[0].ts_since_epoch, 5)
+        self.assertEqual(listens[1].ts_since_epoch, 4)
+        self.assertEqual(listens[2].ts_since_epoch, 3)
+        self.assertEqual(listens[3].ts_since_epoch, 2)
+        self.assertEqual(listens[4].ts_since_epoch, 1)
+
+        self.assert_spark_dump_contains_listens(spark_dump_location, 5)
+        shutil.rmtree(temp_dir)
+
+    def test_full_dump_listen_with_no_insert_timestamp(self):
+        """ We have listens with no `inserted_timestamps` inside the production
+        database. This means that full dumps should always be able to dump these
+        listens as well. This is a test to test that.
+        """
+        listens = generate_data(1, self.testuser_name, 1, 5)
+
+        # insert these listens into influx without an insert_timestamp
+        influx_rows = [listen.to_influx(quote(self.testuser_name)) for listen in listens]
+        for row in influx_rows[1:]:
+            row['fields'].pop('inserted_timestamp')
+
+        t = datetime.now()
+        self.logstore.write_points_to_db(influx_rows)
+        sleep(1)
+        listens_from_influx = self.logstore.fetch_listens(user_name=self.testuser_name, to_ts=11)
+        self.assertEqual(len(listens_from_influx), 5)
+
+        # full dump (with no start time) should contain these listens
+        temp_dir = tempfile.mkdtemp()
+        dump_location = self.logstore.dump_listens(
+            location=temp_dir,
+            dump_id=1,
+            end_time=datetime.now(),
+        )
+        spark_dump_location = self.logstore.dump_listens(
+            location=temp_dir,
+            dump_id=1,
+            end_time=datetime.now(),
+            spark_format=True,
+        )
+        self.assertTrue(os.path.isfile(dump_location))
+        self.reset_influx_db()
+        sleep(1)
+        self.logstore.import_listens_dump(dump_location)
+        sleep(1)
+        listens_from_influx = self.logstore.fetch_listens(user_name=self.testuser_name, to_ts=11)
+        self.assertEqual(len(listens_from_influx), 5)
+        self.assert_spark_dump_contains_listens(spark_dump_location, 5)
+        shutil.rmtree(temp_dir)
+
+    def test_incremental_dumps_listen_with_no_insert_timestamp(self):
+        """ Incremental dumps should only consider listens that have
+        inserted_timestamps.
+        """
+        t = datetime.now()
+        sleep(1)
+        listens = generate_data(1, self.testuser_name, 1, 5)
+
+        # insert these listens into influx without an insert_timestamp
+        influx_rows = [listen.to_influx(quote(self.testuser_name)) for listen in listens]
+        for row in influx_rows[1:]:
```

Comment:
Why is the timestamp being popped off? To make comparisons in the test easier?

Reply:
This is to test that the code doesn't bork with listens that do not have an `inserted_timestamp`. initial releases of LB didn't track when listens were inserted into influx, so we need to take care of that as well.